### PR TITLE
 Ignore `ScheduleSkipRemoveHistory` events

### LINKED ISSUE #3761: chaos-controller-manager: memory keeps increasing
## Bug Report

**What version of Kubernetes are you using?**

**What version of Chaos Mesh are you using?**
 

v2.4.1

**What did you do? / Minimal Reproducible Example**

**What did you expect to see?**

**What did you see instead?**  

img_v2_9c2c1f15-274c-43ea-9286-d1871da3fb4g

```
Controller manager Version: version.Info{GitVersion:"v2.1.4", GitCommit:"7ef267c9e912cdae32dbedc4bd70a4fb546ac3f5", BuildDate:"2022-03-22T07:15:11Z", GoVersion:"go1.17.5", Compiler:"gc", Platform:"linux/amd64"}
2022-11-07T07:17:40.221Z        INFO    fx      fxlog/fxlog.go:65       [Fx] SUPPLY     *prometheus.Registry
2022-11-07T07:17:40.221Z        INFO    fx      fxlog/fxlog.go:65       [Fx] SUPPLY     logr.Logger
2022-11-07T07:17:40.221Z        INFO    fx      fxlog/fxlog.go:57       [Fx] PROVIDE    *metrics.ChaosControllerManagerMetricsCollector <= github.com/chaos-mesh/chaos-mesh/pkg/metrics.NewChaosControllerManagerMetricsCollector()
2022-11-07T07:17:40.221Z        INFO    fx      fxlog/fxlog.go:57       [Fx] PROVIDE    *manager.Options <= github.com/chaos-mesh/chaos-mesh/cmd/chaos-controller-manager/provider.NewOption()
2022-11-07T07:17:40.221Z        INFO    fx      fxlog/fxlog.go:57       [Fx] PROVIDE    client.Client <= github.com/chaos-mesh/chaos-mesh/cmd/chaos-controller-manager/provider.NewClient()
2022-11-07T07:17:40.221Z        INFO    fx      fxlog/fxlog.go:57       [Fx] PROVIDE    *kubernetes.Clientset <= github.com/chaos-mesh/chaos-mesh/cmd/chaos-controller-manager/provider.NewClientSet()
2022-11-07T07:17:40.221Z        INFO    fx      fxlog/fxlog.go:57       [Fx] PROVIDE    manager.Manager <= github.com/chaos-mesh/chaos-mesh/cmd/chaos-controller-manager/provider.NewManager()
2022-11-07T07:17:40.221Z        INFO    fx      fxlog/fxlog.go:57       [Fx] PROVIDE    *v1.AuthorizationV1Client <= github.com/chaos-mesh/chaos-mesh/cmd/chaos-controller-manager/provider.NewAuthCli()
2022-11-07T07:17:40.221Z        INFO    fx      fxlog/fxlog.go:57       [Fx] PROVIDE    *runtime.Scheme <= github.com/chaos-mesh/chaos-mesh/cmd/chaos-controller-manager/provider.NewScheme()
2022-11-07T07:17:40.221Z        INFO    fx      fxlog/fxlog.go:57       [Fx] PROVIDE    *rest.Config <= github.com/chaos-mesh/chaos-mesh/cmd/chaos-controller-manager/provider.NewConfig()
2022-11-07T07:17:40.221Z        INFO    fx      fxlog/fxlog.go:57       [Fx] PROVIDE    client.Reader:no-cache <= github.com/chaos-mesh/chaos-mesh/cmd/chaos-controller-manager/provider.NewNoCacheReader()
2022-11-07T07:17:40.221Z        INFO    fx      fxlog/fxlog.go:57       [Fx] PROVIDE    client.Reader:global-cache <= github.com/chaos-mesh/chaos-mesh/cmd/chaos-controller-manager/provider.NewGlobalCacheReader()
2022-11-07T07:17:40.221Z        INFO    fx      fxlog/fxlog.go:57       [Fx] PROVIDE    client.Reader:control-plane-cache <= github.com/chaos-mesh/chaos-mesh/cmd/chaos-controller-manager/provider.NewControlPlaneCacheReader()
2022-11-07T07:17:40.221Z        INFO    fx      fxlog/fxlog.go:57       [Fx] PROVIDE    *chaosdaemon.ChaosDaemonClientBuilder <= github.com/chaos-mesh/chaos-mesh/controllers/utils/chaosdaemon.New()
2022-11-07T07:17:40.221Z        INFO    fx      fxlog/fxlog.go:57       [Fx] PROVIDE    *recorder.RecorderBuilder <= github.com/chaos-mesh/chaos-mesh/controllers/utils/recorder.NewRecorderBuilder()
2022-11-07T07:17:40.221Z        INFO    fx      fxlog/fxlog.go:57       [Fx] PROVIDE    []pipeline.PipelineStep <= github.com/chaos-mesh/chaos-mesh/controllers/common.AllSteps()
2022-11-07T07:17:40.222Z        INFO    fx      fxlog/fxlog.go:57       [Fx] PROVIDE    *utils.ActiveLister <= github.com/chaos-mesh/chaos-mesh/controllers/schedule/utils.NewActiveLister()
2022-11-07T07:17:40.222Z        INFO    fx      fxlog/fxlog.go:57       [Fx] PROVIDE    *detachvolume.Impl <= github.com/chaos-mesh/chaos-mesh/controllers/chaosimpl/awschaos/detachvolume.NewImpl()
2022-11-07T07:17:40.222Z        INFO    fx      fxlog/fxlog.go:57       [Fx] PROVIDE    *ec2restart.Impl <= github.com/chaos-mesh/chaos-mesh/controllers/chaosimpl/awschaos/ec2restart.NewImpl()
2022-11-07T07:17:40.222Z        INFO    fx      fxlog/fxlog.go:57       [Fx] PROVIDE    *ec2stop.Impl <= github.com/chaos-mesh/chaos-mesh/controllers/chaosimpl/awschaos/ec2stop.NewImpl()
2022-11-07T07:17:40.222Z        INFO    fx      fxlog/fxlog.go:57       [Fx] PROVIDE    *podhttpchaosmanager.Builder <= github.com/chaos-mesh/chaos-mesh/controllers/chaosimpl/httpchaos/podhttpchaosmanager.NewBuilder()
2022-11-07T07:17:40.222Z        INFO    fx      fxlog/fxlog.go:57       [Fx] PROVIDE    *podiochaosmanager.Builder <= github.com/chaos-mesh/chaos-mesh/controllers/chaosimpl/iochaos/podiochaosmanager.NewBuilder()
2022-11-07T07:17:40.222Z        INFO    fx      fxlog/fxlog.go:57       [Fx] PROVIDE    *trafficcontrol.Impl <= github.com/chaos-mesh/chaos-mesh/controllers/chaosimpl/networkchaos/trafficcontrol.NewImpl()
2022-11-07T07:17:40.222Z        INFO    fx      fxlog/fxlog.go:57       [Fx] PROVIDE    *partition.Impl <= github.com/chaos-mesh/chaos-mesh/controllers/chaosimpl/networkchaos/partition.NewImpl()
2022-11-07T07:17:40.222Z        INFO    fx      fxlog/fxlog.go:57       [Fx] PROVIDE    *podnetworkchaosmanager.Builder <= github.com/chaos-mesh/chaos-mesh/controllers/chaosimpl/networkchaos/podnetworkchaosmanager.NewBuilder()
2022-11-07T07:17:40.222Z        INFO    fx      fxlog/fxlog.go:57       [Fx] PROVIDE    *podkill.Impl <= github.com/chaos-mesh/chaos-mesh/controllers/chaosimpl/podchaos/podkill.NewImpl()
2022-11-07T07:17:40.222Z        INFO    fx      fxlog/fxlog.go:57       [Fx] PROVIDE    *podfailure.Impl <= github.com/chaos-mesh/chaos-mesh/controllers/chaosimpl/podchaos/podfailure.NewImpl()
2022-11-07T07:17:40.222Z        INFO    fx      fxlog/fxlog.go:57       [Fx] PROVIDE    *containerkill.Impl <= github.com/chaos-mesh/chaos-mesh/controllers/chaosimpl/podchaos/containerkill.NewImpl()
2022-11-07T07:17:40.222Z        INFO    fx      fxlog/fxlog.go:57       [Fx] PROVIDE    *diskloss.Impl <= github.com/chaos-mesh/chaos-mesh/controllers/chaosimpl/gcpchaos/diskloss.NewImpl()
2022-11-07T07:17:40.222Z        INFO    fx      fxlog/fxlog.go:57       [Fx] PROVIDE    *nodereset.Impl <= github.com/chaos-mesh/chaos-mesh/controllers/chaosimpl/gcpchaos/nodereset.NewImpl()
2022-11-07T07:17:40.222Z        INFO    fx      fxlog/fxlog.go:57       [Fx] PROVIDE    *nodestop.Impl <= github.com/chaos-mesh/chaos-mesh/controllers/chaosimpl/gcpchaos/nodestop.NewImpl()
2022-11-07T07:17:40.222Z        INFO    fx      fxlog/fxlog.go:57       [Fx] PROVIDE    *utils.ContainerRecordDecoder <= github.com/chaos-mesh/chaos-mesh/controllers/chaosimpl/utils.NewContainerRecordDecoder()
2022-11-07T07:17:40.222Z        INFO    fx      fxlog/fxlog.go:57       [Fx] PROVIDE    *selector.Selector <= github.com/chaos-mesh/chaos-mesh/pkg/selector.New()
2022-11-07T07:17:40.222Z        INFO    fx      fxlog/fxlog.go:57       [Fx] PROVIDE    *pod.SelectImpl <= github.com/chaos-mesh/chaos-mesh/pkg/selector/pod.New()
2022-11-07T07:17:40.223Z        INFO    fx      fxlog/fxlog.go:57       [Fx] PROVIDE    *container.SelectImpl <= github.com/chaos-mesh/chaos-mesh/pkg/selector/container.New()
2022-11-07T07:17:40.223Z        INFO    fx      fxlog/fxlog.go:57       [Fx] PROVIDE    *aws.SelectImpl <= github.com/chaos-mesh/chaos-mesh/pkg/selector/aws.New()
2022-11-07T07:17:40.223Z        INFO    fx      fxlog/fxlog.go:57       [Fx] PROVIDE    *gcp.SelectImpl <= github.com/chaos-mesh/chaos-mesh/pkg/selector/gcp.New()
2022-11-07T07:17:40.223Z        INFO    fx      fxlog/fxlog.go:57       [Fx] PROVIDE    *physicalmachine.SelectImpl <= github.com/chaos-mesh/chaos-mesh/pkg/selector/physicalmachine.New()
2022-11-07T07:17:40.223Z        INFO    fx      fxlog/fxlog.go:57       [Fx] PROVIDE    fx.Lifecycle <= go.uber.org/fx.New.func1()
2022-11-07T07:17:40.223Z        INFO    fx      fxlog/fxlog.go:57       [Fx] PROVIDE    fx.Shutdowner <= go.uber.org/fx.(*App).shutdowner-fm()
2022-11-07T07:17:40.223Z        INFO    fx      fxlog/fxlog.go:57       [Fx] PROVIDE    fx.DotGraph <= go.uber.org/fx.(*App).dotGraph-fm()
2022-11-07T07:17:40.223Z        INFO    fx      fx@v1.12.0/app.go:651   [Fx] INVOKE             github.com/chaos-mesh/chaos-mesh/controllers/common.Bootstrap()
2022-11-07T07:17:40.223Z        INFO    setup   provider/controller.go:85       Chaos controller manager is running in cluster scoped mode.
2022-11-07T07:17:40.258Z        INFO    controller-runtime.metrics      logr@v1.2.0/logr.go:249 Metrics server is starting to listen    {"addr": "0.0.0.0:10080"}
2022-11-07T07:17:40.289Z        INFO    setup-common    common/fx.go:72 setting up controller   {"resource-name": "gcpchaos"}
2022-11-07T07:17:40.289Z        INFO    setup-finalizers        finalizers/step.go:32   setting up controller   {"name": "gcpchaos-finalizers"}
2022-11-07T07:17:40.289Z        INFO    setup-desiredphase      desiredphase/step.go:32 setting up controller   {"name": "gcpchaos-desiredphase"}
2022-11-07T07:17:40.289Z        INFO    setup-condition condition/step.go:32    setting up controller   {"name": "gcpchaos-condition"}
2022-11-07T07:17:40.289Z        INFO    setup-common    common/fx.go:72 setting up controller   {"resource-name": "kernelchaos"}
2022-11-07T07:17:40.289Z        INFO    setup-finalizers        finalizers/step.go:32   setting up controller   {"name": "kernelchaos-finalizers"}
2022-11-07T07:17:40.289Z        INFO    setup-desiredphase      desiredphase/step.go:32 setting up controller   {"name": "kernelchaos-desiredphase"}
2022-11-07T07:17:40.289Z        INFO    setup-condition condition/step.go:32    setting up controller   {"name": "kernelchaos-condition"}
2022-11-07T07:17:40.289Z        INFO    setup-common    common/fx.go:72 setting up controller   {"resource-name": "dnschaos"}
2022-11-07T07:17:40.289Z        INFO    setup-finalizers        finalizers/step.go:32   setting up controller   {"name": "dnschaos-finalizers"}
2022-11-07T07:17:40.289Z        INFO    setup-desiredphase      desiredphase/step.go:32 setting up controller   {"name": "dnschaos-desiredphase"}
2022-11-07T07:17:40.289Z        INFO    setup-condition condition/step.go:32    setting up controller   {"name": "dnschaos-condition"}
...skipping...
github.com/chaos-mesh/chaos-mesh/controllers/common/records.(*Reconciler).Reconcile
        /mnt/controllers/common/records/controller.go:85
github.com/chaos-mesh/chaos-mesh/controllers/common/pipeline.(*Pipeline).Reconcile
        /mnt/controllers/common/pipeline/pipeline.go:73
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile
        /tmp/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:114
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
        /tmp/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:311
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
        /tmp/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:266
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
        /tmp/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:227
2022-11-07T07:17:55.931Z        INFO    pipeline        pipeline/pipeline.go:78 reconcile result        {"result": {"Requeue":false,"RequeueAfter":0}}
2022-11-07T07:17:55.931Z        INFO    pipeline        pipeline/pipeline.go:78 reconcile result        {"result": {"Requeue":false,"RequeueAfter":0}}
2022-11-07T07:17:55.931Z        INFO    desiredphase    desiredphase/controller.go:130  modify desiredPhase     {"desiredPhase": "Run"}
2022-11-07T07:17:55.931Z        INFO    pipeline        pipeline/pipeline.go:78 reconcile result        {"result": {"Requeue":false,"RequeueAfter":0}}
2022-11-07T07:17:55.932Z        INFO    pipeline        pipeline/pipeline.go:78 reconcile result        {"result": {"Requeue":false,"RequeueAfter":0}}
2022-11-07T07:17:55.932Z        ERROR   records records/controller.go:85        fail to select  {"error": "no pod is selected"}
github.com/chaos-mesh/chaos-mesh/controllers/common/records.(*Reconciler).Reconcile
        /mnt/controllers/common/records/controller.go:85
github.com/chaos-mesh/chaos-mesh/controllers/common/pipeline.(*Pipeline).Reconcile
        /mnt/controllers/common/pipeline/pipeline.go:73
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile
        /tmp/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:114
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
        /tmp/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:311
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
        /tmp/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:266
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
        /tmp/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:227
2022-11-07T07:17:55.932Z        INFO    pipeline        pipeline/pipeline.go:78 reconcile result        {"result": {"Requeue":false,"RequeueAfter":0}}
2022-11-07T07:17:55.932Z        INFO    pipeline        pipeline/pipeline.go:78 reconcile result        {"result": {"Requeue":false,"RequeueAfter":0}}
2022-11-07T07:17:55.932Z        INFO    desiredphase    desiredphase/controller.go:130  modify desiredPhase     {"desiredPhase": "Run"}
2022-11-07T07:17:55.932Z        INFO    pipeline        pipeline/pipeline.go:78 reconcile result        {"result": {"Requeue":false,"RequeueAfter":0}}
2022-11-07T07:17:55.932Z        INFO    pipeline        pipeline/pipeline.go:78 reconcile result        {"result": {"Requeue":false,"RequeueAfter":0}}
2022-11-07T07:17:55.932Z        ERROR   records records/controller.go:85        fail to select  {"error": "no pod is selected"}
github.com/chaos-mesh/chaos-mesh/controllers/common/records.(*Reconciler).Reconcile
        /mnt/controllers/common/records/controller.go:85
github.com/chaos-mesh/chaos-mesh/controllers/common/pipeline.(*Pipeline).Reconcile
        /mnt/controllers/common/pipeline/pipeline.go:73
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile
        /tmp/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:114
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
        /tmp/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:311
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
        /tmp/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:266
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
        /tmp/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:227
2022-11-07T07:17:55.932Z        INFO    pipeline        pipeline/pipeline.go:78 reconcile result        {"result": {"Requeue":false,"RequeueAfter":0}}
2022-11-07T07:17:55.932Z        INFO    pipeline        pipeline/pipeline.go:78 reconcile result        {"result": {"Requeue":false,"RequeueAfter":0}}
2022-11-07T07:17:55.932Z        INFO    desiredphase    desiredphase/controller.go:130  modify desiredPhase     {"desiredPhase": "Run"}
2022-11-07T07:17:55.932Z        INFO    pipeline        pipeline/pipeline.go:78 reconcile result        {"result": {"Requeue":false,"RequeueAfter":0}}
2022-11-07T07:17:55.932Z        INFO    pipeline        pipeline/pipeline.go:78 reconcile result        {"result": {"Requeue":false,"RequeueAfter":0}}
2022-11-07T07:17:55.932Z        ERROR   records records/controller.go:85        fail to select  {"error": "no pod is selected"}
github.com/chaos-mesh/chaos-mesh/controllers/common/records.(*Reconciler).Reconcile
        /mnt/controllers/common/records/controller.go:85
github.com/chaos-mesh/chaos-mesh/controllers/common/pipeline.(*Pipeline).Reconcile
        /mnt/controllers/common/pipeline/pipeline.go:73
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile
        /tmp/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:114
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
        /tmp/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:311
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
        /tmp/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:266
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
        /tmp/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:227
2022-11-07T07:17:55.932Z        INFO    pipeline        pipeline/pipeline.go:78 reconcile result        {"result": {"Requeue":false,"RequeueAfter":0}}
I1107 07:17:56.886350       1 request.go:665] Waited for 1.010160615s due to client-side throttling, not priority and fairness, request: POST:https://172.20.0.1:443/api/v1/namespaces/chaos-testing/events
I1107 07:18:06.911412       1 request.go:665] Waited for 10.031230553s due to client-side throttling, not priority and fairness, request: POST:https://172.20.0.1:443/api/v1/namespaces/chaos-testing/events
I1107 07:18:16.947103       1 request.go:665] Waited for 19.064485164s due to client-side throttling, not priority and fairness, request: POST:https://172.20.0.1:443/api/v1/namespaces/chaos-testing/events
I1107 07:18:26.977577       1 request.go:665] Waited for 27.089388658s due to client-side throttling, not priority and fairness, request: POST:https://172.20.0.1:443/api/v1/namespaces/chaos-testing/events
I1107 07:18:36.978383       1 request.go:665] Waited for 35.085107146s due to client-side throttling, not priority and fairness, request: POST:https://172.20.0.1:443/api/v1/namespaces/chaos-testing/events
``` 
**Output of chaosctl**

   

pprof: 

image
image

### PR DESCRIPTION
**Pull request by cwen0**
Signed-off-by: cwen0 <cwenyin0@gmail.com>

### What problem does this PR solve?

 

Fix #3761 

### What's changed and how it works?

 

* Ignore `ScheduleSkipRemoveHistory` events

### Related changes

- [ ] This change also requires further updates to the [website](https://github.com/chaos-mesh/website) (e.g. docs)
- [ ] This change also requires further updates to the `UI interface`
- Need to **cheery-pick to release branches**
  - [x] release-2.4
  - [ ] release-2.3

### Checklist

CHANGELOG

- [ ] I have updated the `CHANGELOG.md`
- [ ] I have labeled this PR with "no-need-update-changelog"

Tests

- [ ] Unit test
- [ ] E2E test
- [ ] No code
- [x] Manual test (add steps below)

Side effects

- [ ] **Breaking backward compatibility**

### DCO

If you find the DCO check fails, please run commands like below (Depends on the actual situations. For example, if the failed commit isn't the most recent) to fix it:

```shell
git commit --amend --signoff
git push --force
```

### CODE REVIEW

File: controllers/schedule/gc/controller.go [hunk 1]
```diff
@@ -90,13 +92,9 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
...
 				if !finished {
 					if untilStop != 0 {
-						if requeuAfter == 0 || requeuAfter > untilStop {
+						if untilStop>requeuAfter {
 							requeuAfter = untilStop
 						}
-
-						r.Recorder.Event(schedule, recorder.ScheduleSkipRemoveHistory{
-							RunningName: innerObj.GetName(),
-						})
```
Comment (STRRL) [inline controllers/schedule/gc/controller.go:99]: after we removed these lines to emit event, do we still need to change `requeueAfter` to 10 mins ?

Comment (cwen0) [reply controllers/schedule/gc/controller.go:99]: It is not necessary. I will reset it.

Comment (cwen0) [reply controllers/schedule/gc/controller.go:99]: Updated

**Review by STRRL — APPROVED** [review]
LGTM

**Review by YangKeao — APPROVED** [review]
LGTM

**Pull request merged**